lib/mongo/utils.rb

```diff
@@ -103,6 +103,7 @@ def initialize(**opts)
     end
 
     # Hash#compact implementation for Ruby 2.3
+    # @deprecated Use stdlib version since min Ruby version required is 2.4.
```

Comment:
This entire module is private, so we can remove this method at any time. If you don't want to change calls throughout the tree to use Hash#compact now would you make a new ticket to do that please?

Reply:
I think it's better to remove such code in this PR. Will do.